core/src/main/scala/chisel3/internal/firrtl/Converter.scala

```diff
@@ -410,7 +410,7 @@ private[chisel3] object Converter {
           fir.Field(getRef(elt, info).name, fir.Flip, extractType(elt, false, info, checkProbe, true, typeAliases))
       }
       if (!t._isOpaqueType)
-        fir.BundleType(t._elements.toIndexedSeq.reverse.map { case (_, e) => eltField(e) })
+        fir.BundleType(t._elements.toIndexedSeq.map { case (_, e) => eltField(e) })
```

Comment:
you can't make this change here because it is affecting all Records, not just Bundles. So even if you undo the reversing for Bundles, this change is now messing with records so .asUint on a record would have different behavior.

YOu could show this by taking the cookbook here https://www.chisel-lang.org/docs/cookbooks/cookbook#how-do-i-create-a-bundle-from-a-uint and rewriting it with a record and convincing yourself it doesn't work the same after this change